### DISCUSSION TITLE
ui-selected doesn't work as expected with nested components (combobox in tab)

### DISCUSSION THREAD

**Discussion by R4VANG3R** [tailwindlabs/headlessui#2366] [Help]
As said in the title, I'm using a combobox in a tab (both of which support the ui-selected class for tailwindcss). 

I'm finding that when I nest them like this the combobox's options (where I use ui-selected) are always selected. I'm guessing they're inheriting this from the Tab, but this isn't what I would expect.

Is there a way to fix this?

**Comment by gztomas** [comment]
I'm having the same exact issue

**Comment by gztomas** [comment]
A workaround I found is adding `ui-open`, for instance:

```
ui-open:ui-selected:font-bold
```

Another alternative for things that are `static` is adding `ui-not-selected`

```
ui-selected:font-bold ui-not-selected:font-normal
```

**Comment by lilkimo** [comment]
Same here, ui-selected works weird when there are nested components, such as nested tabs.

**Comment by mnlfischer** [comment]
The issue here is that for example the Listbox picks up the `data-headlessui-state` from the parent tab state and all Listbox options are now active.
